projects/aca-folder-rules/src/lib/rules-list/rule/rule-list-item.ui-component.ts

```diff
@@ -34,4 +35,18 @@ import { Rule } from '../../model/rule.model';
 export class RuleListItemUiComponent {
   @Input() rule: Rule;
   @Input() isSelected: boolean;
+  @Input() nodeId: string;
+
+  constructor(private folderRulesService: FolderRulesService) {}
+
+  onToggleClick() {
+    this.folderRulesService.toggleRule(this.nodeId, this.rule.id, this.parseRule(this.rule));
+  }
+
+  private parseRule(rule): Rule {
+    const parsedRule = { ...rule, enabled: !this.rule.enabled };
```

Comment:
Could we try and obtain this value directly from the mat-slide-toggle rather than just negating it?